src/write.rs

```diff
@@ -281,6 +281,14 @@ impl<W: Write> Drop for BzDecoder<W> {
     }
 }
 
+impl<W: Write> Drop for BzEncoder<W> {
+    fn drop(&mut self) {
+        if self.obj.is_some() {
+            let _ = self.try_finish();
```

Comment:
`std::io::BufWriter` skips the flush on Drop if a panic happened while writing to the inner writer. It does so by having a `panicked` field which is set to true before writing to the inner writer and set to false immediately after writing. This avoids a double panic if the inner writer would panic on the second write too.

Reply:
Are you implying that we should be doing something similar here?

Reply:
It would be nice to do so yeah.

Reply:
I do not understand what you want me to tell here or how to apply the `panicked` field in this implementation.